setup.py

```diff
@@ -33,38 +33,36 @@
         "Operating System :: MacOS :: MacOS X",
     ],
     install_requires=[
-        'imageio',
-        'matplotlib',
-        'numpy',
-        'pandas',
-        'scipy',
-        'toml',
-        'tqdm',
+        "imageio",
+        "matplotlib",
+        "numpy",
+        "pandas",
+        "scipy",
+        "sklearn",
+        "toml",
+        "tqdm",
+        "xgboost",
```

Comment:
Does xgboost need to be are requirement for the minimal installation. Feels like it rather  belongs to examples, or does it not?

Reply:
I make xgboost optional now and set the default to not run it.